c/serverStatusService.c

```diff
@@ -0,0 +1,272 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+
+#ifdef METTLE
+#include <metal/metal.h>
+#include <metal/stddef.h>
+#include <metal/stdio.h>
+#include <metal/stdlib.h>
+#include <metal/string.h>
+#include "metalio.h"
+#include "qsam.h"
+#else
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#endif
+#include <sys/utsname.h>
+
+#include "zowetypes.h"
+#include "bpxnet.h"
+#include "utils.h"
+#include "socketmgmt.h"
+
+#include "logging.h"
+#include "zssLogging.h"
+#include "serverStatusService.h"
+
+#ifdef __ZOWE_OS_ZOS
+static int serveStatus(HttpService *service, HttpResponse *response);
+extern char **environ;
+
+typedef int EXSMFI(int *reqType, int *recType, int *subType,
+                   char* buffer, int *bufferLen, int *cpuUtil,
+                   int *dpRate, int *options, int *mvs, int *zaap, int *ziip);
+EXSMFI *smfFunc;
+
+int installServerStatusService(HttpServer *server, JsonObject *serverSettings, char* productVer)
```

Comment:
Can this be void? You return zero regardless, since there is no failure condition.